### PR TITLE
Split truthy into sub-packages

### PR DESCRIPTION
**Pull request by jhbforlife**
I was excited to find this package! I would find it very useful to have this type of functionality so thought I would add it.

### CODE REVIEW

**Comment by earthboundkid** [comment]
Thanks. I have another package with pointer.Deref. I wonder if they should just be combined? Seems related.

**Comment by earthboundkid** [comment]
https://pkg.go.dev/github.com/carlmjohnson/pointer#Deref

**Comment by jhbforlife** [comment]
I definitely think that could work! I can mess around with what that could look like later today

**Comment by jhbforlife** [comment]
@carlmjohnson just pushed an update! I may have gone overboard but I actually kind of like having everything separated like this - it makes it clearer to me what exactly I'm trying to use

File: is/is.go [hunk 1]
```diff
@@ -25,36 +32,33 @@ func ValueAny[T any](v T) bool {
...
-// ValueMap returns true if the length of the map is greater than 0.
+// TruthyMap returns true if the length of the map is greater than 0.
 // Note that it does not distinguish nil maps from empty maps.
-func ValueMap[K comparable, V any, M ~map[K]V](v M) bool {
+func TruthyMap[K comparable, V any, M ~map[K]V](v M) bool {
 	return len(v) > 0
 }
 
-// Value returns the truthy value of comparable types.
+// Get returns the truthy value of dereferenced pointers of comparable types.
```
Comment (earthboundkid) [inline is/is.go]: This should be "TruthyPointer returns".

File: is/is.go [hunk 2]
```diff
@@ -25,36 +32,33 @@ func ValueAny[T any](v T) bool {
...
+	if v == nil {
+		return false
+	}
+	return Truthy(*v)
+}
+
+// TruthySlice returns true if the length of the slice is greater than 0.
+// Note that it does not distinguish nil slices from empty slices.
+func TruthySlice[T any, S ~[]T](v S) bool {
+	return len(v) > 0
```
Comment (earthboundkid) [inline is/is.go]: `is.TruthySlice(s)` is longer than `len(s) > 0`. Is the clarity worth it? Maybe if it were a little shorter?

File: defaults/defaults.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package defaults
+
+import "github.com/carlmjohnson/truthy/is"
+
+// GetFirst returns the first value in vs which is non-zero.
+func GetFirst[T comparable](vs ...T) (t T) {
```
Comment (earthboundkid) [inline defaults/defaults.go]: Go 1.22 will have cmp.Or (which I added 😊) that does this same thing, so it's kind of redundant to have it here. OTOH, FirstAny is still useful.

**Comment by earthboundkid** [comment]
What if the subpackages were bools and pointers? The idea is the standard library has strings, bytes, slices, maps… and we're adding bools and pointers. The bools package would have stuff about truthiness and pointers would be what you have here.

**Comment by earthboundkid** [comment]
A bools package could also have bools.Int like this proposal: https://github.com/golang/go/issues/64825.

File: is/is.go (hunk 2 above)
Comment (jhbforlife) [reply is/is.go]: what do you think about [e66db9d](https://github.com/carlmjohnson/truthy/pull/2/commits/e66db9d72ad3de562654c0c4a46df1ed265f469b)?

File: defaults/defaults.go (hunk 3 above)
Comment (jhbforlife) [reply defaults/defaults.go]: hmm... could it just be removed once it's officially live in Go? that's pretty sweet though

**Review by earthboundkid — COMMENTED** [review]
Overall, it looks good, but since this is supposed to just be a convenience thing, I think we should push the names to be short as we can without losing clarity. Name suggestions:

- Rename package pointers to ptrs. I think ptr is used as a variable too much to be a package name, but ptrs should be free to use, and it's much shorter.
- bools.Comparable to bools.Value.
- bools.GetFirst and bools.GetFirstAny to bools.Or and bools.AnyOr.
- bools.SetFirst and SetFirstAny to bools.Set and bools.AnySet.
- bools.Ternary to bools.If.

What do you think?

**Pull request closed without merging**